pkg/kubeapiserver/admission/configuration/configuration_manager.go

```diff
@@ -81,9 +93,25 @@ func (a *poller) notReady() {
 	a.ready = false
 }
 
+func (a *poller) timeoutBootstrap() {
+	// bootstrapGracePeriod is read-only, so no lock is required
+	timer := time.NewTimer(a.bootstrapGracePeriod)
+	go func() {
+		<-timer.C
+		a.lock.Lock()
+		defer a.lock.Unlock()
+		a.bootstrapped = true
+		a.bootstrappedCondition.Broadcast()
+	}()
+}
+
 func (a *poller) configuration() (runtime.Object, error) {
-	a.lock.RLock()
-	defer a.lock.RUnlock()
+	a.once.Do(a.timeoutBootstrap)
+	a.lock.Lock()
+	defer a.lock.Unlock()
+	if !a.bootstrapped {
+		a.bootstrappedCondition.Wait()
```

Comment:
Wait() will not return until the thread wins a.lock, so the piled up requests are not resumed at the same time. If we want to further spread them, i can let the thread sleep for 0.1s after Wait() returns, though i don't think it's necessary.

BTW, does the initialization AC get called before the generic webhook AC? If that's the case, the generic webhook AC doesn't need the protection during start-up, since the initialization AC has already delayed the AC chain.